src/main/webapp/app/core/main/controllers/productController.js

```diff
@@ -20,10 +20,49 @@ app.controller("ProductCtrl", function ($scope, $uibModal, $http, toastr,$locati
         ProductService.getAllProducts()
             .then(function successCallback(response) {
                 productCtrl.productsList = response.data;
+                checksProductsValidity();
             }, function errorCallback(error) {
             });
     };
-    
+
+    var checksProductsValidity = function () {
+        var batchesByProduct = [];
+
+        _.forEach(productCtrl.productsList, function (product) {
+            ProductService.getBatchesByProduct(product.barCode)
+                .then(function successCallback(response) {
+                    batchesByProduct = response;
+                    productCtrl.productsBatches[product.barCode] = batchesByProduct;
+                    checksForAllBatches(product);
+                    updateProduct(product);
+                }, function errorCallback(error) {
+                });
+        });
+    };
+
+    var checksForAllBatches = function (product) {
+        _.forEach(productCtrl.productsBatches[product.barCode], function (batch) {
+            if (batch.expirationDate < todaysDate() && product.quantity > 0)
+                product.quantity -= batch.numberOfItems;        
+        });
+        if (product.quantity <= 0) product.statusCode = 2;
```

Comment:
there is a constant in the constants.js that has all the product status properties. So you can't set to 2, set it to constant.AVAILABLE.value